### PR TITLE
Adding a check of mechanisms settings.

### PR DESCRIPTION
**Pull request by fbacchella**
Try to resolve issue #641, by putting checks when binding or connecting for all security mechanisms.

### CODE REVIEW

File: src/main/java/zmq/io/mechanism/Mechanisms.java [hunk 1]
```diff
@@ -36,6 +53,22 @@ public Mechanism create(SessionBase session, Address peerAddress, Options option
...
+        @Override
+        public void check(Options options)
+        {
+            if ((options.curvePublicKey == null ^ options.curveSecretKey == null)) {
+                throw new IllegalStateException("Curve mechanism definition incomplete");
+            }
+            if (options.curvePublicKey != null && options.curvePublicKey.length != Options.CURVE_KEYSIZE) {
+                throw new IllegalStateException("Curve mechanism definition incomplete");
+            }
+            if (options.curveSecretKey != null && options.curveSecretKey.length != Options.CURVE_KEYSIZE) {
```
Comment (fredoboulo) [inline src/main/java/zmq/io/mechanism/Mechanisms.java]: As a user, I would be welcoming all options config problems at once, the proposed algo means that if 2 keys are misconfigured, you will have to try 2 times to find out all the problems.

**Comment by fbacchella** [comment]
openjdk build failed on , oraclejdk succeeded. It that really linked to my code ?

**Comment by fredoboulo** [comment]
I got similar problem on my latest PR. Timeout on travis-ci after 10 mins. I assume it's not related to your change, but rather to infrastructure issues

**Comment by daveyarwood** [comment]
In my experience, force-pushing can cause issues with Travis. The best thing to do when that happens is to retry the build in the Travis UI.

**Comment by fredoboulo** [comment]
@daveyarwood where in Travis UI can we retry the build?

**Comment by fbacchella** [comment]
perhaps raising this ZMQ.sleep(1):
https://github.com/zeromq/jeromq/blob/master/src/test/java/zmq/TestTermEndpoint.java#L236
would help.
Travis is very sensible to timing problem, as it can be very slow.

**Comment by daveyarwood** [comment]
Sleeping in tests is an antipattern in general. Perhaps we should be checking the assertion in a loop and failing only if the assertion doesn't hold after a timeout (e.g. 5 seconds).

**Comment by fredoboulo** [comment]
The test is all about disconnecting an endpoint and ensuring that it is not possible to send messages anymore. AND the disconnect process is asynchronous.
Best way we could do would be to monitor the socket and wait for the disconnection event... But it's not straightforward in zmq namespace.

**Comment by trevorbernard** [comment]
This is a difficult problem since the disconnects are async and the monitor wasn't designed for flow/state control. It was a mechanism for logging and debugging.

**Comment by daveyarwood** [comment]
I've written tests like this before -- I could take a stab at removing reliance on "sleep" in our tests sometime.

**Comment by fbacchella** [comment]
Does it need to be in the critical path of this PR (or a 0.5 too) ?

**Comment by fredoboulo** [comment]
No, it's not related to this PR. It just happened that it triggered the problem.

**Pull request merged**